### PR TITLE
Cleanup selection logic and throttle some events

### PR DESCRIPTION
**Pull request by martinRenou**
The main view would always deselect the current selection without checking that it needs to do it, this PR fixes it

### CODE REVIEW

File: src/mainview.tsx [hunk 1]
```diff
@@ -69,6 +69,25 @@ interface IPickedResult {
...
+        callback(...args);
+      }, delay);
+    } else {
+      last = now;
+      callback(...args);
+    }
+  } as T;
+}
+
+
```
Comment (trungleduc) [inline src/mainview.tsx:90]: I would put it in the `tools.ts` file.

**Review by hbcarlos — APPROVED** [review]
Thanks @martinRenou

**Comment by hbcarlos** [comment]
Can you add `jupyter_ydoc==0.2.2` as a dependency [here](https://github.com/martinRenou/jupytercad/blob/7dc74b041e2969b5d76da557de9011a7e757c515/pyproject.toml#L10)?
That might fix the tests.

**Comment by martinRenou** [comment]
Thanks Carlos!

**Pull request merged**